src/View/Helper/HtmlHelper.php

```diff
@@ -232,7 +232,11 @@ public function meta($type, $content = null, array $options = [])
             ];
 
             if ($type === 'icon' && $content === null) {
-                $types['icon']['link'] = 'favicon.ico';
```

Comment:
Just replacing `'favicon.ico'` here with `$this->Url->assetUrl('favicon.ico', ['pathPrefix' => ''])` should do the job.